src/util.cc

```diff
@@ -2550,6 +2550,24 @@ TEST_CASE("util split") {
     }
 }
 
+TEST_CASE("util approx_equal") {
+    CHECK(approx_equal(47.0, 47.0) == true);
+    CHECK(approx_equal(47.0, -47.0) == false);
+    CHECK(approx_equal(47.00001, 47.00002) == false);
+    CHECK(approx_equal(47.00001, 47.00002, 1e-5) == true);
+    CHECK(approx_equal(47.0, -47.0, 1e2) == true);
+    CHECK(approx_equal(47.0, -47.0, 94 + 1e-10) == true);
+    CHECK(approx_equal(47.0, -47.0, 94) == false);
+    CHECK(approx_equal(1 / 0.0, 2 / 0.0) == false);
+    CHECK(approx_equal(1 / 0.0, 2 / 0.0, 3 / 0.0) == false);
+    CHECK(approx_equal(0.0 / 0.0, 0.0 / 0.0) == false);
```

Comment:
The MSVC compiler will actually complain about divide-by-zeroes by default:

```
FAILED: src/CMakeFiles/zeek_objs.dir/util.cc.obj 
"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\Hostx86\x64\cl.exe"  /nologo /TP -DCARES_STATICLIB -DDOCTEST_CONFIG_NO_MULTITHREADING -DDOCTEST_CONFIG_SUPER_FAST_ASSERTS -DZEEK_CONFIG_SKIP_VERSION_H -DZEEK_PLUGIN_SKIP_VERSION_CHECK -Isrc\include -Isrc -I..\src\include -I..\src -I. -I..\auxil\out_ptr\include -I..\auxil\rapidjson\include -I..\auxil\paraglob\include -I..\auxil\highwayhash -Iauxil\broker\libbroker -I..\auxil\broker\libbroker -I"C:\Program Files\OpenSSL-Win64\include" -Ivcpkg_installed\x64-windows-static\include -Iauxil\binpac\lib -I..\auxil\binpac\lib -I..\auxil\libunistd\regex -I..\auxil\libunistd\unistd /DWIN32 /D_WINDOWS /GR /EHsc /O2 /Ob2 /DNDEBUG /MT -MT /guard:cf /Z7 /J /wd4068 -std:c++17 /showIncludes /Fosrc\CMakeFiles\zeek_objs.dir\util.cc.obj /Fdsrc\CMakeFiles\zeek_objs.dir\ /FS -c ..\src\util.cc
..\src\util.cc(2561): error C2124: divide or mod by zero
```

I believe there's a checker for ubsan for it too, but we don't enable ubsan jobs for PR builds.

Reply:
so should we just leave those (and the NaNs) out? Certainly seems simplest

Reply:
One can explicitly construct infinity and NaN values so we could (and should) leave in a test, e.g.,

```cpp
constexpr auto inf = std::numeric_limits<double>::infinity();
CHECK_FALSE(approx_equal(inf, inf));
CHECK_FALSE(approx_equal(-inf, inf));
CHECK_FALSE(approx_equal(inf, -inf));
CHECK_FALSE(approx_equal(inf, inf, inf));

constexpr auto qnan = std::numeric_limits<double>::quiet_NaN(); // There's also `signalining_NaN`.
CHECK_FALSE(approx_equal(qnan, qnan));
CHECK_FALSE(approx_equal(-qnan, qnan));
CHECK_FALSE(approx_equal(qnan, -qnan));
```